### PR TITLE
[QPE-290] Null values selection prevention

### PR DESCRIPTION
**Pull request by ahmed-Bazzara**
Jira ticket link 

https://tretton37.atlassian.net/secure/RapidBoard.jspa?rapidView=33&projectKey=QPE&modal=detail&selectedIssue=QPE-290

### CODE REVIEW

**Comment by ahmed-Bazzara** [comment]
i added Dim1IsNull and Dim2IsNull properties to the Data array,, then made a check on both of them in order to prevent the null selection.
It might be that one of these properties in enough,, a decision left to you @kristofferlind

File: src/paint.js [hunk 1]
```diff
@@ -129,12 +129,26 @@ function convertHYPERCUBEtoJSON(layout) {
     var dim1Id = qMatrix.map(function(d) {
       return d[0].qElemNumber;
     });
+    var dim1IsNull = qMatrix.map(function(d) {
```
Comment (kristofferlind) [inline src/paint.js]: rework this a bit so its just defined once per dimension

File: src/paint.js [hunk 2]
```diff
@@ -129,12 +129,26 @@ function convertHYPERCUBEtoJSON(layout) {
...
+        return false;
+      }
+    });
     var dim2Labels = qMatrix.map(function(d) {
       return d[1].qText;
     });
     var dim2Id = qMatrix.map(function(d) {
       return d[1].qElemNumber;
     });
+    var dim2IsNull = qMatrix.map(function(d) {
```
Comment (kristofferlind) [inline src/paint.js]: figure out a more suitable name for these, dim2IsNull suggests it has to do with whether or not the dimension is null whereas this is an index or lookup table for whether or not a specific entry is null.

File: src/paint.js [hunk 3]
```diff
@@ -195,7 +216,7 @@ function convertHYPERCUBEtoJSON(layout) {
       // it is a different grouping value of Dim1
       LegendValues.push(dim1Labels[k]);
       // Make sure radar_area is added for usage in the radar chart layers later
-      myJson.definition[cont] = { "axis_id" : dim2Id[k], "axis" : dim2Labels[k], "radar_area_id" : dim1Id[k], "radar_area" : dim1Labels[k], "value" : metric1Values[k] };
+      myJson.definition[cont] = { "axis_id" : dim2Id[k], "axis" : dim2Labels[k], "dim2IsNull": dim2IsNull[k], "radar_area_id" : dim1Id[k], "radar_area" : dim1Labels[k], "dim1IsNull": dim1IsNull[k], "value" : metric1Values[k] };
```
Comment (kristofferlind) [inline src/paint.js]: definition is defined 3 times and they're identical

**Pull request merged**